src/cljs/main/broadfcui/page/workspace/data/import_data.cljs

```diff
@@ -69,17 +69,22 @@
    :do-upload
    (fn [{:keys [props state]}]
      (swap! state assoc :loading? true)
-     (endpoints/call-ajax-orch
-      {:endpoint ((if (= "data" (:import-type props))
-                    endpoints/import-entities
-                    endpoints/import-attributes)
-                  (:workspace-id props))
-       :raw-data (utils/generate-form-data {(if (= "data" (:import-type props)) :entities :attributes) (:file @state)})
-       :encType "multipart/form-data"
-       :on-done (fn [{:keys [success? get-parsed-response]}]
-                  (swap! state dissoc :loading? :file :file-contents)
-                  (if success?
-                    (do
-                      (swap! state assoc :upload-result {:success? true})
-                      ((:on-data-imported props)))
-                    (swap! state assoc :upload-result {:success? false :error (get-parsed-response false)})))}))})
+     (if-let [{:keys [on-upload]} props]
```

Comment:
The internet promised me that destructuring in a `if-let` would work but it doesn't. 😠 Fix coming.